src/main/java/picard/arrays/illumina/CreateExtendedIlluminaManifest.java

```diff
@@ -0,0 +1,701 @@
+                sb.append("NOTES / Warnings:\n");
+                if (!creator.isRefStrandDefinedInManifest()) {
+                    sb.append("REF_STRAND was NOT defined in the manifest.  We have inferred it from sequence / strand information in the manifest.\n");
+                }
+                if (!creator.getUnsupportedBuilds().isEmpty()) {
+                    sb.append("Records were found within the manifest on Genome Builds for which you have not provided liftover information.\n");
+                    sb.append(" They have been failed with the flag: UNSUPPORTED_GENOME_BUILD\n");
+                    sb.append(" The following unexpected Genome Builds were found: ").append(StringUtils.join(creator.getUnsupportedBuilds(), ", ")).append("\n");
+                }
+                sb.append("\n");
+            }
+            manifestStatistics.logStatistics(REPORT_FILE, sb.toString());
+        } catch (IOException e) {
+            throw new PicardException(e.getMessage(), e);
+        }
+
+        return 0;
+    }
+
+    /**
+     * This method goes through a list of records and if it finds duplicates, it flags the non-best
+     * with the duplicate flag.
+     *
+     * It iterates over all of the records, looking for records that share the same locus (chrom/position) and alleles.
+     * If it finds more than one record that share these attributes, it will flag all but one as 'duplicate'
+     *  this flag will later be used to set the 'DUPE' filter in the GtcToVcf tool.
+     * The decision as to which record will be flagged as duplicate is based upon the GenTrain score
+     *  (cluster quality from the GenTrain clustering algorithm) which is pulled from the Illumina cluster file.
+     * The record with the highest GenTrain score will NOT be flagged as a duplicate, all others will.
+     *
+     * @param records A list of records to search through and flag for duplicates.
+     */
+    private Set<Integer> flagDuplicates(final List<Build37ExtendedIlluminaManifestRecord> records) {
+        // Load the cluster file to get the GenTrain scores
+        // load the egt first, and create a map of ilmnid to gentrain score.  Save that and use it for deduplicating.
+        log.info("Loading the egt file for duplicate resolution");
+        final InfiniumEGTFile infiniumEGTFile;
+        try {
+            infiniumEGTFile = new InfiniumEGTFile(CLUSTER_FILE);
+        } catch (IOException e) {
+            throw new PicardException("Error reading cluster file '" + CLUSTER_FILE.getAbsolutePath() + "'", e);
+        }
+        final Map<String, Float> nameToGenTrainScore = new HashMap<>();
+        for (String rsName : infiniumEGTFile.rsNameToIndex.keySet()) {
+            nameToGenTrainScore.put(rsName, infiniumEGTFile.totalScore[infiniumEGTFile.rsNameToIndex.get(rsName)]);
+        }
+
+        return flagDuplicates(records, nameToGenTrainScore);
```

Comment:
Not sure I understand.  I'm returning the Set<Integer> which is changed by the method that's called here (it has a different signature, and I've only done it this way so that I could expose the functionality of this method for testing.

Reply:
Great, I think i was just missing where it is changed.